core/scripts/vrfv1/main.go

```diff
@@ -25,6 +25,17 @@ func panicErr(err error) {
 	}
 }
 
+func failIfRequiredArgumentsAreEmpty(required []string) {
```

Comment:
A better way I think would be to pass the [`FlagSet`](https://pkg.go.dev/flag#FlagSet) object into this method and call [`Visit`](https://pkg.go.dev/flag#FlagSet.Visit) on the flagset itself rather than via `flag.Visit`. E.g:

```golang
func parseArgs(fset *flag.FlagSet, args []string, requiredArgs []string) {
  panicErr(fset.Parse(args))
  seen := map[string]bool{}
  fset.Visit(func(f *flag.Flag) { seen[f.Name] = true })
  for _, req := range required {
    if !seen[req] {
      panicErr(fmt.Errorf("missing required -%s argument/flag", req))
    }
  }
}
```

Then we can call it like this:

```golang
case "ownerless-consumer-deploy":
  cmd := flag.NewFlagSet("ownerless-consumer-deploy", flag.ExitOnError)
  // ... defining flags on cmd here ...
  // parse and enforce presence of flags
  parseArgs(cmd, os.Args[2:], cmd.Args())
```

Reply:
I was super tempted to switch these commands to Cobra (https://github.com/spf13/cobra) which would handle required args and subcommands much better (no need for switch-case)..wonder if it's worth it if we expect external devs to use these scripts.

Reply:
Unfortunately to add cobra means another dependency for the whole app, though I think we can use https://github.com/urfave/cli which is what we use for the chainlink binary CLI.

Reply:
Even I was tempted to use a lib to validate arguments, yes with that we need to manage one more dependency. I believe to manage the scripts long-term it makes sense to use a lib either cobra or urfave. Shall we move ahead with urfave ?

Reply:
Sounds good to me!

Reply:
Yup sounds good!

Reply:
@Nic0S @makramkd Apparently github.com/urfave/cli doesnt support required args. I'm just fixing up the PR with the comments. I believe this should be good for now.